### ISSUE TITLE
Artemis Update to support null-safety and be compatible with other packages.

### ISSUE THREAD

**Issue by inc16sec** [comigor/artemis#279]
I'm using Flutter 2.0 and Dart sdk: ">=2.7.0 <3.0.0" and wanted to migrate my project to make it null-safe, but in order to do so I must first update all of the packages that I'm using to a version that supports null-safety (including this one). When I update packages like "gql" OR  "graphql-flutter" an error shows up

```
Because artemis >=6.17.2 depends on gql ^0.12.4 and uth_dev_02 depends on gql ^0.13.0-nullsafety.2, artemis >=6.17.2 is forbidden.

So, because uth_dev_02 depends on artemis ^7.0.0-beta.1, version solving failed.
pub get failed (1; So, because uth_dev_02 depends on artemis ^7.0.0-beta.1, version solving failed.)
exit code 1
```
So I'm wondering if there's something I need to do from my side or I must just wait for an update that supports null-safety and can be compatible with the other packages. 
Best regards.

**Comment by michael-woolsey** [comment]
I understand that `7.0.0-beta.1` supports the null safety, the `Min Dart SDK` is still saying only 2.8 instead of the 2.12 and I suspect the `dart pub outdated --mode=null-safety` command needs to find 2.12 to be marked as null safe.

```
dev_dependencies:
artemis                            ✗6.18.4        ✗6.18.4        -           ✗7.0.0-beta.1
build_runner                       ✗1.10.12       ✗1.10.12       -           ✗1.12.2
```

However that's a guess on my part as I'm not a package developer. :)

**Comment by inc16sec** [comment]
> I understand that `7.0.0-beta.1` supports the null safety, the `Min Dart SDK` is still saying only 2.8 instead of the 2.12 and I suspect the `dart pub outdated --mode=null-safety` command needs to find 2.12 to be marked as null safe.
> 
> ```
> dev_dependencies:
> artemis                            ✗6.18.4        ✗6.18.4        -           ✗7.0.0-beta.1
> build_runner                       ✗1.10.12       ✗1.10.12       -           ✗1.12.2
> ```
> 
> However that's a guess on my part as I'm not a package developer. :)

What I've done is, #comment the package from `pubspec.yaml` and then after migrating I installed it again. But still you will incounter some slight issues when trying to build and you gonna have to override some dependencies till the package gets updated with the correct version for those dependecies that you've overriden. check [https://github.com/comigor/artemis/pull/259#issuecomment-810716278](https://github.com/comigor/artemis/pull/259#issuecomment-810716278) for more.

**Comment by comigor** [comment]
@inc16sec  We've just merged https://github.com/comigor/artemis/pull/283 with NNBD support for Artemis.
I'm closing this but feel free to open another issue, if applicable, after testing the new version `7.0.0-beta.2`.

**Comment by inc16sec** [comment]
> @inc16sec We've just merged #283 with NNBD support for Artemis.
> I'm closing this but feel free to open another issue, if applicable, after testing the new version `7.0.0-beta.2`.

Just tested it and it works as expected with no issues.
Thank you all for the great job you're doing and for taking the time to make sure this package keeps going further ❤️.
